x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/geo/GeoShape.java

```diff
@@ -41,41 +48,49 @@
 
     public static final String NAME = "geo";
 
-    private final ShapeBuilder<?, ?, ?> shapeBuilder;
+    private final Geometry shape;
 
     public GeoShape(double lon, double lat) {
-        shapeBuilder = new PointBuilder(lon, lat);
+        shape = new Point(lat, lon);
     }
 
     public GeoShape(Object value) throws IOException {
-        shapeBuilder = ShapeParser.parse(value);
+        try {
+            shape = parse(value);
+        } catch (ParseException ex) {
+            throw new ElasticsearchParseException("cannot load shape", ex);
```

Comment:
Why do you need to wrap it in a new exception here? And if it's the case shouldn't this be an `Sql` exception?
Also can we have some tests to check those cases?

Reply:
> Why do you need to wrap it in a new exception here? And if it's the case shouldn't this be an Sql exception?

I wrapped it here for consistency with the next constructor which cannot throw anything besides IOException, I will rewrap the it into `SqlIllegalArgumentException` that sounds like a better idea.

> Also can we have some tests to check those cases?

We are already testing this cases in [`StWkttosqlProcessorTests`](https://github.com/elastic/elasticsearch/blob/08da5bb5eadd0efd79ba5fad325da29219b89623/x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/expression/function/scalar/geo/StWkttosqlProcessorTests.java#L35).